core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -187,7 +187,7 @@ class TransactionCoordinator(brokerId: Int,
     } else {
       // caller should have synchronized on txnMetadata already
       txnMetadata.state match {
-        case PrepareAbort | PrepareCommit =>
+        case PrepareAbort | PrepareCommit | Dead =>
```

Comment:
related discussions on the previous PR: https://github.com/apache/kafka/pull/3101#discussion_r118397818